sidechain/consensus/aura/src/lib.rs

```diff
@@ -245,7 +246,7 @@ where
 		.current_validateers(header)
 		.map_err(|e| ConsensusError::CouldNotGetAuthorities(e.to_string()))?
 		.into_iter()
-		.map(|e| AuthorityId::<P>::from_slice(e.pubkey.as_ref()))
+		.filter_map(|e| AuthorityId::<P>::from_slice(e.pubkey.as_ref()).ok())
```

Comment:
What has changed here? Does `AuthorityId::<T>::from_slice()` now return a `Result<R>` and before it just returned `R`?

Reply:
The reason is that with AccountId not offering a Default anymore, ValueQueries in Pallets (returned Default upon not having an entry inside storage) needed to be changed to OptionQueries - Returning a None value in case the storage is emtpy instead of a Default.

Relevant PR: https://github.com/paritytech/substrate/pull/10403

Reply:
Alright, thanks for the explanation 👍

Reply:
Sorry, actually the above was wrong - you were right: It returns a Result, not an Option.

As written in https://github.com/paritytech/substrate/pull/10403:
> The function ByteArray::from_slice (which used to be Public::from_slice) has been changed to return a Result in the case that the slice provided was the incorrect size. For an infallible conversion, use the UncheckedFrom trait with a fixed size array.